docs/tutorial/prismic-source-graphql-tutorial.md

```diff
@@ -0,0 +1,333 @@
+
+You will have to setup your `gatsby-browser.js` file to use the Link Resolver as well.
+
+```javascript:title=gatsby-browser.js
+
+const { registerLinkResolver } = require('gatsby-source-prismic-graphql');
+const { linkResolver } = require('./src/utils/linkResolver');
+  
+registerLinkResolver(linkResolver);
+```
+
+You can now use the proper url generated by the `linkResolver` function to build the `<Link>` component for each blog post, so that that post titles are links now.
+
+```javascript:title=src/pages/index.js
+
+const BlogPosts = ({ posts }) => {
+  if(!posts) return null;
+  return(
+    <div>
+      {posts.map((post) => {
+        return(
+          <div key={post.node.id}>
+            // highlight-start
+            <Link to={ linkResolver(post.node._meta) }>
+              { RichText.asText(post.node.title) }
+            </Link>
+            // highlight-end
+            <p>
+              <time>{post.node.date}</time>
+            </p>
+          </div>
+        );
+      })}
+    </div>
+  );
+}
+```
+
+At this point, if you try to use these links to navigate your site, you will notice that you're faced with a 404 error on every page. That's because even though you're navigating to the pages correctly, there's still the issue of actually creating the pages. For this, you will have to use a template file which will programmatically build a page for each blog post.
+
+### Creating pages programmatically with a template
+
+The template you will build is very similar in concept to a regular page, with the addition of introducing a variable which will be used to query the different blog posts.
+
+```javascript:title=/src/templates/post.js
+import React from 'react'
+import { graphql, Link } from 'gatsby'
+import { RichText } from 'prismic-reactjs'
+
+export const query = graphql`
+query BlogPostQuery($uid: String) {
+  prismic {
+    allPosts(uid: $uid) {
+      edges {
+        node{
+          title
+          date
+          post_body
+        }
+      }
+    }
+  }
+}`
+
+export default ({ data }) => {
+  const doc = data.prismic.allPosts.edges.slice(0,1).pop();
+  if (!doc) return null;
+
+  return (
+    <div>
+      <Link to="/">
+        <span>go back</span>
+      </Link>
+      <h1>{RichText.asText(doc.node.title)}</h1>
+      <span><em>{doc.node.date}</em></span>
+      <div>
+        {RichText.render(doc.node.post_body)}
+      </div>
+    </div>
+  );
+}
+```
+
+We're rendering a very minimalistic page for each blog post, consisting of a link back to the homepage, the article's title and page, and finally the rich text body which will contain formatted text and images. Having a template is not enough to generate the dynamic pages, you will need to configure the plugin in `gatsby-config.js` so that pages are created following the same pattern defined in the link resolver function
```

Comment:
```suggestion
We're rendering a very minimalistic page for each blog post, consisting of a link back to the homepage, the article's title and page, and finally the rich text body which will contain formatted text and images. Having a template is not enough to generate the dynamic pages, you will need to configure the plugin in `gatsby-config.js` so that pages are created following the same pattern defined in the link resolver function:
```